ui/packages/code/interactive/Code.svelte

```diff
@@ -0,0 +1,194 @@
+
+	async function get_lang(val: string) {
+		const ext = await getLanguageExtension(val);
+		lang_extension = ext;
+	}
+
+	$: reconfigure(), lang_extension;
+	$: setDoc(value);
+
+	function setDoc(newDoc: string) {
+		if (view && newDoc !== view.state.doc.toString()) {
+			view.dispatch({
+				changes: {
+					from: 0,
+					to: view.state.doc.length,
+					insert: newDoc
+				}
+			});
+		}
+	}
+
+	function createEditorView(): EditorView {
+		return new EditorView({
+			parent: element,
+			state: createEditorState(value)
+		});
+	}
+
+	function handleChange(vu: ViewUpdate): void {
+		if (vu.docChanged) {
+			const doc = vu.state.doc;
+			const text = doc.toString();
+			value = text;
+			dispatch("change", text);
+		}
+	}
+
+	function getExtensions() {
+		const stateExtensions = [
+			...getBaseExtensions(
+				basic,
+				useTab,
+				placeholder,
+				readonly,
+				lang_extension
+			),
+			FontTheme,
+			...getTheme(),
+			...extensions
+		];
+		return stateExtensions;
+	}
+
+	const FontTheme = EditorView.theme({
+		"&": {
+			fontSize: "var(--text-sm)",
+			backgroundColor: "var(--color-border-secondary)"
+		},
+		".cm-content": {
+			paddingTop: "5px",
+			paddingBottom: "5px",
+
+			fontFamily: "var(--font-mono)",
+			minHeight: "100%"
+		},
+		".cm-gutter": {
+			minHeight: "231px"
+		},
+		".cm-gutters": {
+			marginRight: "1px",
+			borderRight: "1px solid var(--color-border-primary)",
+			backgroundColor: "transparent",
+			color: "var(--text-color-subdued)"
+		},
+		".cm-focused": {
+			outline: "none"
+		}
+	});
+
+	function createEditorState(value: string | null | undefined): EditorState {
+		return EditorState.create({
+			doc: value ?? undefined,
+			extensions: getExtensions()
+		});
+	}
+
+	function getBaseExtensions(
+		basic: boolean,
+		useTab: boolean,
+		placeholder: string | HTMLElement | null | undefined,
+		readonly: boolean,
+		lang: Extension | null | undefined
+	): Extension[] {
+		const extensions: Extension[] = [
+			EditorView.editable.of(!readonly),
+			EditorState.readOnly.of(readonly)
+		];
+
+		if (basic) {
+			extensions.push(basicSetup);
+		}
+		if (useTab) {
+			extensions.push(keymap.of([indentWithTab]));
+		}
+		if (placeholder) {
+			extensions.push(placeholderExt(placeholder));
+		}
+		if (lang) {
+			extensions.push(lang);
+		}
+
+		extensions.push(EditorView.updateListener.of(handleChange));
+		return extensions;
+	}
+
+	function getTheme(): Extension[] {
+		const extensions: Extension[] = [];
+
+		if (dark_mode) {
+			extensions.push(basicDark);
+		} else {
+			extensions.push(basicLight);
+		}
+		return extensions;
+	}
+
+	function reconfigure(): void {
+		view?.dispatch({
+			effects: StateEffect.reconfigure.of(getExtensions())
+		});
+	}
+
+	onMount(() => {
+		view = createEditorView();
+		return () => view?.destroy();
+	});
+</script>
+
+<div class="wrap">
+	<div class="codemirror-wrapper {classNames}" bind:this={element} />
+</div>
+
+<style>
+	.codemirror-wrapper {
+		padding-top: 25px;
```

Comment:
In the theme PR, I changed how we handle the space for the BlockLabel. Instead of having to provide padding-top/margin-top in the nested component, just set the `<BlockLabel float={false}>` This makes `position: static` for BlockLabel, so the nested component is automatically positioned underneath. Otherwise, if the user changes the font or padding of BlockLabel, this specific padding will not be sufficient

Reply:
This is applied to the code editor itself. I don't want to float the block label, I want the code to scroll underneath the label because it looks far cleaner. So it needs to be absolutely positioned. I'll change it to rems, so it scales with font size but I don't know how to account for padding applied elsewhere.

Reply:
fixed this now.